src/app/SharpRaven/Utilities/RuntimeInfoHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Reflection;
+#if HAS_RUNTIME_INFORMATION
+using System.Runtime.InteropServices;
+#elif NET45
+using Microsoft.Win32;
+#endif
+
+namespace SharpRaven.Utilities
+{
+    internal static class RuntimeInfoHelper
+    {
+        public static string GetRuntimeVersion()
+        {
+#if HAS_RUNTIME_INFORMATION
+            // Prefered API: netstandard2.0 and vNext
+            // https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs
+            // e.g: .NET Framework 4.7.2633.0, .NET Native, WebAssembly
+            var version = RuntimeInformation.FrameworkDescription;
+#elif NET45
+            var mono = GetFromMono();
+            var version = mono
+                // Windows only: Get latest installation of .NET Framework 4.5 or later
+                ?? Get45PlusLatestInstallationFromRegistry();
+#else
+            var mono = GetFromMono();
+            var version = mono
+                // Environment.Version: NET Framework 4, 4.5, 4.5.1, 4.5.2 = 4.0.30319.xxxxx
+                // .NET Framework 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1 =  4.0.30319.42000
+                // Not recommended on NET45+
+                ?? $".NET Framework {Environment.Version}";
+#endif
+            return version;
+        }
+
+        private static string GetFromMono()
+        {
+            string monoVersion = null;
+
+            var type = Type.GetType("Mono.Runtime", false);
+            if (type != null)
+            {
+                var displayName = type.GetMethod("GetDisplayName", BindingFlags.NonPublic | BindingFlags.Static);
+                if (displayName != null)
+                {
+                    monoVersion = displayName.Invoke(null, null) as string;
+                    if (monoVersion != null)
```

Comment:
Can't all of these `null` checks be collapsed into a chained `null` propagation (`?.`) expression?

Reply:
I was afraid it'd be less readable but it's fine. Will change.